modules/spotxBidAdapter.js

```diff
@@ -0,0 +1,357 @@
+
+          const bid = {
+            requestId: request.video.ext.bidId,
+            currency: serverResponseBody.cur || 'USD',
+            cpm: spotxBid.price,
+            creativeId: spotxBid.crid || '',
+            ttl: 360,
+            netRevenue: true,
+            channel_id: serverResponseBody.id,
+            cache_key: spotxBid.ext.cache_key,
+            slot: request.video.ext.slot
+          };
+
+          if (request.video) {
+            bid.vastUrl = '//search.spotxchange.com/ad/vast.html?key=' + spotxBid.ext.cache_key;
+            bid.mediaType = VIDEO;
+            bid.width = spotxBid.w;
+            bid.height = spotxBid.h;
+          }
+
+          const videoMediaType = utils.deepAccess(bidderRequest, 'mediaTypes.video');
+          const context = utils.deepAccess(bidderRequest, 'mediaTypes.video.context');
+          if ((bid.mediaType === 'video' || (videoMediaType && context !== 'outstream')) || (request.video.ext.ad_unit == 'outstream')) {
+            const renderer = Renderer.install({
+              id: 0,
+              url: '//',
+              config: {
+                adText: 'SpotX Outstream Video Ad via Prebid.js',
+                player_width: request.video.ext.player_width,
+                player_height: request.video.ext.player_height,
+                content_page_url: request.video.ext.content_page_url,
+                ad_mute: request.video.ext.ad_mute,
+                outstream_options: request.video.ext.outstream_options,
+                outstream_function: request.video.ext.outstream_function
+              }
+            });
+
+            try {
+              renderer.setRender(outstreamRender);
+              renderer.setEventHandlers({
+                impression: function impression() {
+                  return utils.logMessage('SpotX outstream video impression event');
+                },
+                loaded: function loaded() {
+                  return utils.logMessage('SpotX outstream video loaded event');
+                },
+                ended: function ended() {
+                  utils.logMessage('SpotX outstream renderer video event');
+                }
+              });
+            } catch (err) {
+              utils.logWarn('Prebid Error calling setRender or setEve,tHandlers on renderer', err);
+            }
+            bid.renderer = renderer;
+          }
+
+          bidResponses.push(bid);
+        })
+      });
+    }
+
+    return bidResponses;
+  }
+}
+
+function outstreamRender(bid) {
+  if (bid.renderer.config.outstream_function != null && typeof bid.renderer.config.outstream_function === 'function') {
+    bid.renderer.config.outstream_function(bid);
+  } else {
+    try {
+      utils.logMessage('[SPOTX][renderer] Handle SpotX outstream renderer');
+      const slot = utils.getBidIdParameter('slot', bid.renderer.config.outstream_options);
+      const inIframe = utils.getBidIdParameter('in_iframe', bid.renderer.config.outstream_options);
```

Comment:
Can you document this parameter? From my understanding this should be true if DFP is used as it always delivers creatives in iframes.

Reply:
If customers decide to render inside an iframe then they will be able to use this function. Depending on use cases it can be used or not. We'll document it in the helper file.